src/nvim/extmark.c

```diff
@@ -394,14 +394,13 @@ void extmark_apply_undo(ExtmarkUndoObject undo_info, bool undo)
     ExtmarkSavePos pos = undo_info.data.savepos;
     if (undo) {
       if (pos.old_row >= 0) {
-        extmark_setraw(curbuf, pos.mark, pos.old_row, pos.old_col);
-      }
-      if (pos.invalidated) {
-        MarkTreeIter itr[1] = { 0 };
-        MTKey mark = marktree_lookup(curbuf->b_marktree, pos.mark, itr);
-        mt_itr_rawkey(itr).flags &= (uint16_t) ~MT_FLAG_INVALID;
-        MTPos end = marktree_get_altpos(curbuf->b_marktree, mark, itr);
-        buf_put_decor(curbuf, mt_decor(mark), mark.pos.row, end.row);
+        if (extmark_setraw(curbuf, pos.mark, pos.old_row, pos.old_col) && pos.invalidated) {
+          MarkTreeIter itr[1] = { 0 };
+          MTKey mark = marktree_lookup(curbuf->b_marktree, pos.mark, itr);
+          mt_itr_rawkey(itr).flags &= (uint16_t) ~MT_FLAG_INVALID;
+          MTPos end = marktree_get_altpos(curbuf->b_marktree, mark, itr);
+          buf_put_decor(curbuf, mt_decor(mark), mark.pos.row, end.row);
+        }
       }
       // Redo
```

Comment:
@bfredl should this branch be removed or may it be used in the future? Currently `pos.row` always equals `-1`.

Reply:
it's fine to delete it.